File: non-duplicate-array/NonDuplicateArray.js
function nonDuplicate1(array) {
    const result = [];
    array.sort((a, b) => a - b);
    for(let i = 0; i < array.length; i++) {
        let distinctIndex = i;
        let isDuplicated = false;
        for(let j = i + 1; j < array.length; j++) {
            if(isDuplicated && array[distinctIndex] !== array[j]) { break; }
            if(array[distinctIndex] === array[j]) {
                distinctIndex = j;
                isDuplicated = true;
            }
        }
        result.push(array[distinctIndex]);
        i = distinctIndex;
    }
    return result;
}

function nonDuplicate2(array) {
    var set = new Set();
    let index = 0;
    while(index < array.length) {
        set.add(array[index]);
        index++;
    }
    return [...set]; //Array.from(set)
}

exports.nonDuplicate = nonDuplicate1

File: non-duplicate-array/NonDuplicateArray.cs
using System;
using System.Collections.Generic;

public class NonDuplicateArray {
    public static int[] NonDuplicate(int[] array) {
        Array.Sort(array);
        List<int> results = new List<int>();
        for (int i = 0; i < array.Length; i++)
        {
            int distinctIndex = i;
            bool isDuplicated = false;
            for (int j = i + 1; j < array.Length; j++)
            {
                if(isDuplicated && array[distinctIndex] != array[j]) { break; }
                if (array[distinctIndex] == array[j])
                {
                    distinctIndex = j;
                    isDuplicated = true;
                }
            }
            results.Add(array[distinctIndex]);
            i = distinctIndex;
        }
        return results.ToArray();
    }

    public static int[] NonDuplicate2(int[] array) {
        HashSet<int> set = new HashSet<int>();
        int index = 0;
        while(index < array.Length) {
            set.Add(array[index]);
            index++;
        }
        return set.ToArray();
    }

    public static int[] NonDuplicate3(int[] array) {
        HashSet<int> set = new HashSet<int>(array); return set.ToArray();
    }
}